diff --git a/pkg/reconciler/eventlistener/resources/config.go b/pkg/reconciler/eventlistener/resources/config.go
index 20b4bfc6..1d5efc4a 100644
--- a/pkg/reconciler/eventlistener/resources/config.go
+++ b/pkg/reconciler/eventlistener/resources/config.go
@@ -39,6 +39,7 @@ var (
 	DefaultStaticResourceLabels = map[string]string{
 		"app.kubernetes.io/managed-by": "EventListener",
 		"app.kubernetes.io/part-of":    "Triggers",
+		"kcp.dev/cluster":              "local",
 	}
 	// DefaultSystemNamespace is the default system namespace used.
 	DefaultSystemNamespace = "tekton-pipelines"
